pulsar/internal/rpc_client.go

```diff
@@ -105,6 +110,27 @@ func (c *rpcClient) Request(logicalAddr *url.URL, physicalAddr *url.URL, request
 	}
 }
 
+func (c *rpcClient) getConn(logicalAddr *url.URL, physicalAddr *url.URL,
+	connectingThroughProxy bool) (Connection, error) {
+	cnx, err := c.pool.GetConnection(logicalAddr, physicalAddr, connectingThroughProxy)
+	backoff := new(Backoff)
+	var retryTime time.Duration
+	if err != nil {
+		for retryTime < c.requestTimeout {
```

Comment:
Will this retry forever?

Reply:
the `OperationTimeout` in ClientOptions struct (default: 30 seconds)